util/checker/src/cmd/check_stubs_compile.rs

```diff
@@ -0,0 +1,169 @@
+use std::{fs, path::PathBuf, process::Command};
+
+fn make_reserve_copies(modified_exercise: &PathBuf) {
+    let stub_path = modified_exercise.join("src").join("lib.rs");
+
+    let stub_path_copy = modified_exercise.join("src").join("lib.rs.orig");
+
+    let tests_path = modified_exercise.join("tests");
+
+    let tests_path_copy = modified_exercise.join("tests.orig");
+
+    fs::copy(&stub_path, &stub_path_copy)
+        .unwrap_or_else(|_| panic!("Failed to make a reserve copy for {:?}", &stub_path));
+
+    if !tests_path_copy.exists() {
+        fs::create_dir(&tests_path_copy)
+            .unwrap_or_else(|_| panic!("Failed to create {:?} directory", &tests_path_copy));
+    }
+
+    for tests_entry in fs::read_dir(&tests_path)
+        .unwrap_or_else(|_| panic!("Failed to read {:?} directory", &tests_path))
+    {
+        if let Ok(tests_entry) = tests_entry {
+            let entry_path = tests_entry.path();
+            fs::copy(
+                &entry_path,
+                &tests_path_copy.join(entry_path.file_name().unwrap()),
+            ).unwrap_or_else(|_| panic!("Failed to make a reserve copy for {:?}", entry_path));
+        }
+    }
+}
+
+fn remove_copies(modified_exercise: &PathBuf) {
+    let stub_path = modified_exercise.join("src").join("lib.rs");
```

Comment:
Given that you have to produce the mapping of originals and copies in a few places, it seems like it may be worthwhile to extract out a common function which generates a list of tuples, which you could iterate over.